x-pack/plugins/upgrade_assistant/public/application/components/es_deprecations/deprecation_tab_content.tsx

```diff
@@ -100,28 +155,92 @@ export const DeprecationTabContent: FunctionComponent<CheckupTabProps> = ({
   if (isLoading) {
     content = <SectionLoading>{i18nTexts.isLoading}</SectionLoading>;
   } else if (deprecations?.length) {
+    const levelGroups = groupBy(deprecations, 'level');
+    const deprecationLevelsCount = Object.keys(levelGroups).reduce((counts, level) => {
+      counts[level] = levelGroups[level].length;
+      return counts;
+    }, {} as { [level: string]: number });
```

Comment:
NIT. Might be easier to read

```suggestion
    }, {} as Record<string, number>);
```